### DISCUSSION TITLE
Optimizing the image instead of the model weights

### DISCUSSION THREAD

**Discussion by mrhoads14** [open-mmlab/mmdetection#8567] [Q&A]
I'd like to use mmdetection models to optimize (learn) the pixel values for an image. This means I'd like to keep the weights of the detection model fixed and take the gradient of the loss with respect to the input image pixels. Is there a way that I can get the gradient of a loss function with respect to the input image pixel values using mmdetection models?

I know how to do this via pytorch because one can use a tensor throughout and simply ensure `requires_grad=True` on the input image tensor. After finding the loss, one can then call `loss.backward()` and then the input image tensor will have it's `grad `attribute populated. However, because mmdetection models take ndarrays as input and output, I'm unsure how to do this with mmdetection models. Any help is appreciated!
